cpp/src/arrow/compute/util.cc

```diff
@@ -58,6 +58,13 @@ void TempVectorStack::release(int id, uint32_t num_bytes) {
   --num_vectors_;
 }
 
+void TempVectorStack::CheckAllocSizeValid(int64_t estimate_alloc_size) {
```

Comment:
`CheckOverflow()`, `CheckEnoughSize()` or something?

Reply:
Changed to CheckAllocOverflow() ?